core/raftlease/fsm.go

```diff
@@ -101,35 +107,46 @@ func (f *FSM) extend(key lease.Key, holder string, duration time.Duration) *resp
 	return &response{}
 }
 
+func (f *FSM) pin(key lease.Key) *response {
+	f.pinned[key] = true
+	return &response{}
+}
+
+func (f *FSM) unpin(key lease.Key) *response {
+	delete(f.pinned, key)
+	return &response{}
```

Comment:
Do we want to expire any leases that would have expired while the pin was in place? I guess just leaving it until the next time update is fine though.

Reply:
I think it is worth leaving for now and revisiting when we think more about a mechanism for preventing instant lease expiry when leases are unpinned.